week_3_alt_assignment/index.js

```diff
@@ -0,0 +1,307 @@
+              with!
+            </p>
+
+            Then, <a href="/whatisaserver/dynamicHtml/extra">next demo</a>
+        `
+        : ""
+    }
+    </body></main>
+  `);
+});
+
+app.get("/whatisaserver/dynamicHtml/extra", (request, response) => {
+  response.send(`
+    ${STYLE}
+    <body><main>
+    <p>
+      With a little creativity, it is amazing how much we can build with
+      these simple primitives. For example:
+      <ul>
+        <a href="/whatisaserver/dynamicHtml/extra?lightsOn=true"><li>Lights Off</li></a>
+        <a href="/whatisaserver/dynamicHtml/extra?lightsOn=false"><li>Lights On</li></a>
+        <a href="/whatisaserver/dynamicHtml/extra"><li>Undo</li></a>
+      </ul>
+    </p>
+
+    <!-- And here are the styles and server-side code to enable the above
+    trickery to work. Remember the ${"javascript"} code runs on the server! --!>
+
+    ${
+      // I am using strings that say "true" and "false" which is a bit
+      // confusing. These are strings, not boolean values! Hence, I actually
+      // have a lot more than 2 possibilities; I only turn the lights on
+      // if I have an exact "true". I turn them off for an exact "false".
+      // Otherwise, I do nothing!
+      request.query["lightsOn"] === "true"
+        ? "<style>body {background-color: black }</style>"
+        : request.query["lightsOn"] === "false"
+        ? "<style>body { background-color: #ffff82 /* light yellow */ }</style>"
+        : ""
+    }
+
+
+    <h2>Pretty cool!</h2>
+    <p><a href="/whatisaserver/json">Next Demo</a></p>
+    </body></main>
+  `);
+});
+
+app.get("/whatisaserver/json", (_, response) => {
+                                // ^ It is not a bad idea to use an underscore
+                                // to displace positional arguments that you
+                                // don't need. I don't need `request`, so I
+                                // just put an _. This is not very beautiful,
+                                // but sometimes developers do this if they
+                                // have no other choice!
+  response.contentType("application/json").send(`
+    {
+      "Oh!": "and don't get stressed about JSON!",
+      "it": [
+        "turns",
+        "out",
+        "it's",
+        "just",
+        "another",
+        "type",
+        "of",
+        "text",
+        "that",
+        "HTTP",
+        "servers",
+        "like",
+        "to",
+        "send!"
+      ],
+      "and": {
+        "if": "you are a human user, it might not look that pretty,",
+        "but": "if you are a computer, this stuff is absolutely infectious!"
+      },
+      "finalPageHref": "http://localhost:3000/whatisaserver/xmlcloser",
+      "note": "visit the above link next!     ^^^^^^^^^^^^^^^^^^^^^^^"
+    }
+  `);
+});
+
+// XML is an older alternative to JSON, used by many old and crusty enterprise
+// software systems. Hopefully this will be the first and last time you
+// see it :)
```

Comment:
I get what you're saying :sweat\_smile:  but I probably wouldn't put it that way myself. 

Yes often more likely to see it in older / enterprise systems but still, I wouldn't dismiss it as crusty. It may be useful for students to know that

*   the relationship between XML / HTML (they may notice that it looks similar. They both descended from `Standard Generalized Markup Language` but have different use cases)
*   you may see it in payloads for things like ecommerce, banking, sports betting, weather 
*   It is considered more human-readable vs JSON being targeted to machines (though of course with practice, JSON is plenty human-readable)
*   you can add comments and attributes to tags which you can't really do in standard JSON
*   if they ever go into mobile development, XML is often used for creating layouts in Android development (I'm not familiar with iOS development but maybe there too)
*   may be seen in some configuration files instead of JSON
*   can be converted to JSON and vice-versa: https://codebeautify.org/xmltojson